eng/build.ps1

```diff
@@ -505,6 +505,47 @@ try {
     if ($ci) {
         List-Processes
         Prepare-TempDir
+
+        if ($testVsi) {
+            $screenshotPath = (Join-Path $LogDir "StartingBuild.png")
+            try {
+                Capture-Screenshot $screenshotPath
+            }
+            catch {
+                Write-Host "Screenshot failed; attempting to connect to the console"
+
+                # Keep the session open so we have a UI to interact with
+                $quserItems = ((quser $env:USERNAME | select -Skip 1) -split '\s+')
+                $sessionid = $quserItems[2]
+                if ($sessionid -eq 'Disc') {
+                    # When the session isn't connected, the third value is 'Disc' instead of the ID
+                    $sessionid = $quserItems[1]
+                }
+
+                if ($quserItems[1] -eq 'console') {
+                    Write-Host "Disconnecting from console before attempting reconnection"
+                    try {
+                        tsdiscon
+                    } catch {
+                        # ignore
+                    }
+
+                    # Disconnection is asynchronous, so wait a few seconds for it to complete
+                    Start-Sleep -Seconds 3
+                    query user
+                }
+
+                Write-Host "tscon $sessionid /dest:console"
+                tscon $sessionid /dest:console
+
+                # Connection is asynchronous, so wait a few seconds for it to complete
+                Start-Sleep 3
+                query user
```

Comment:
This is an EXE whose failures will be ignored. Is that intended?

Reply:
Yes, it just shows the output in the build log.

Reply:
➡️ Follow-up pull request #33800 submitted